### ISSUE TITLE
Web UI should allow access to model training/validation metrics in real time

### ISSUE THREAD

**Issue by mameshini** [kubeflow/pipelines#2416]
Currently, TensorBoard button is only available in Pipelines UI at the end of each experiment run.  

As a data scientist performing model training, I want to access to TensorBoard immediately after the start of the pipeline step, so I can monitor taining/validation metrics and often I may want to abort an experiment based on model metrics.  TensorBoard is logging training and validation metrics such as error, loss, and accuracy.  These metrics have to be used in real time to diagnose everything from vanishing gradients to exactly when models start overfitting.  There is no need to wait extra hours to complete the run, if it does not help to improve the model.

This issue is related to:
https://github.com/kubeflow/pipelines/issues/1497

TensorBoard data should be stored in a volume, and Pipelines UI can provide a button to TensorBoard once the data is available without waiting to complete the pipeline step.

**Comment by IronPan** [comment]
I think real time rendering would be nice for volume as well as for GCS or S3.

**Comment by Mddct** [comment]
any update on prem k8s ?

**Comment by Bobgy** [comment]
/reopen
If real time metrics are required, but KFP cannot currently support it.
Will you consider building your own server that collects real time metrics?
What's missing there that blocks satisfying your users?
